app/controllers/accounts_controller.rb

```diff
@@ -65,36 +65,18 @@ def logout
 
 
   def get_profile
-    if @current_user
-      user = User.find_by(id: params[:id].to_i)
-
-      unless user
-        render(inertia: 'NotFound')
-        return
-      end
-
-      user_json = user.as_json(only: [:id, :name, :username, :description, :avatar])
-
-      student_reviews = Review.where(connection_id: Connection.where(match_id: MatchRequest.where(student_id: user.id)))
-      teacher_reviews = Review.where(connection_id: Connection.where(match_id: MatchRequest.where(skill_class_id: SkillClass.where(teacher_id: user.id))))
+    user = User.find_by(id: params[:id].to_i)
 
-      total_reviews = student_reviews.or(teacher_reviews)
-
-      avg_rating = total_reviews.average(:rating).to_f.round(1) # only one decimal case
+    unless user
+      render(inertia: 'NotFound')
+      return
+    end
 
-      if @current_user.id == user.id
-        #own profile, can edit
-        res = {"can_edit": true, "user": user_json, "rating": avg_rating}
+    user_json = ProfileInfo.new(user).get
```

Comment:
This looks amazing. You nailed it 🙌 Just me being picky now. json translation should be outside ProfileInfoService. Just because it's sort of a formatting step which is only useful on the controller response.